src/main/java/org/opensearch/knn/quantization/models/quantizationParams/ScalarQuantizationParams.java

```diff
@@ -36,6 +37,31 @@ public static String generateTypeIdentifier(ScalarQuantizationType sqType) {
         return generateIdentifier(sqType.getId());
     }
 
+    public ScalarQuantizationParams(ScalarQuantizationType quantizationType) {
```

Comment:
Do we need all of these constructors?  We can still use all arg constructor and then use force = true?

Reply:
I think we probably should switch to a lombok builder here for the following reasons:
1. disambiguate the serialization/deserialization constructors from "business logic" `enableADC/enableRandomRotation` constructors
2. we now have multiple boolean values so maintainer could make mistake down the line if they misorder them
3. allows us to keep `enableADC/enableRandomRotation` as `final` variables without needing explicit constructors.

What do you think?